tap_klaviyo/__init__.py

```diff
@@ -172,29 +174,32 @@ def get_available_metrics(api_key):
     return metric_streams
 
 
-def discover(api_key):
-    metric_streams = get_available_metrics(api_key)
+def discover(headers):
+    metric_streams = get_available_metrics(headers)
     return {"streams": [a.to_catalog_dict()
                         for a in metric_streams + FULL_STREAMS]}
 
 
-def do_discover(api_key):
-    print(json.dumps(discover(api_key), indent=2))
+def do_discover(headers):
+    print(json.dumps(discover(headers), indent=2))
 
 @singer.utils.handle_top_exception(LOGGER)
 def main():
 
     args = singer.utils.parse_args(REQUIRED_CONFIG_KEYS)
+    headers = {
+        "Authorization": f"Klaviyo-API-Key {args.config.get('api_key')}",
+        "revision": f"{args.config.get('api_version', API_VERSION)}"
```

Comment:
AFAIK, API_VERSION is hardcoded value in the code and should not be taken from the config.

Reply:
Fixed it.